intermediate.atx.2017Q1.Arnold.Wells

Arnold73 opened the pull request:
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of JavaScript: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*






## Checkpoint 2: Towers of Hanoi GUI App
- [ ] All tests passed: 55 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of jQuery: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*






## Checkpoint 3: Final Project Gist Blog
- [x] Effective use of AJAX: 20 points
- [x] Effective use of loops: 20 points
- [x] Effective use of github API: 20 points
- [ ] Style and design: 10 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

Arnold73 committed:
changed to true

Arnold73 committed:
homework complete

Arnold73 committed:
Rock Paper Scissors

Arnold73 committed:
lesson 2 homework turn in

Arnold73 committed:
got it all worked out now

nevetsretlaw commented on apps/02RockPaperScissors.js:
```diff
@@ -10,8 +10,38 @@ var rl = readline.createInterface({
-  // Write code here
+  // hand1 and hand are just variables
 
+if (hand1 === hand2) {
```
Lines 14-44 should be indented one because they are part of the larger function

Arnold73 committed:
I was able to figure it out

Arnold73 committed:
homework 3 turn in.

Arnold73 committed:
barebones TicTacToe

Arnold73 committed:
homework 4 complete

pieroapretto commented on apps/03TicTacToe.js:
```diff
@@ -24,23 +24,68 @@ function printBoard() {
+      checkForWin();
+      playerTurn = 'O';
+    } 
+    else {
```
You can use the value of playerTurn to appropriately assign either 'X' or 'O' to the board.

This will save you from having to write an `if\else` statement.

Arnold73 committed:
finished this

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -19,23 +19,33 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
```
We need to get some comments in here.

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -19,23 +19,33 @@ function printStacks() {
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
+  isLegal(startStack, endStack);
+    checkForWin();
```
Make sure everything within a block of curly braces is aligned together

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -19,23 +19,33 @@ function printStacks() {
-
+function isLegal(startStack, endStack) {
+  if  (stacks[startStack[startStack.length - 1]] < stacks[endStack[endStack.length - 1]] || stacks[endStack].length === 0) {
+    movePiece(startStack, endStack);
```
This does work but see if you can find a way to code it differently without having movePiece() inside of isLegal

Arnold73 committed:
final turn in

Arnold73 commented:
i made the changes.  added comments. cleaned up indentation, and moved the
move piece function into the main function.

Arnold

On Tue, Feb 14, 2017 at 10:16 PM, Jon Woo <notifications@github.com> wrote:

> *@jw56578* commented on this pull request.
> ------------------------------
>
> In apps/04TowersOfHanoi.js
> <https://github.com/AustinCodingAcademy/javascript-workbook/pull/150#pullrequestreview-21921302>
> :
>
> >  }
>
> -function isLegal() {
> -  // Your code here
> -
> +function isLegal(startStack, endStack) {
> +  if  (stacks[startStack[startStack.length - 1]] < stacks[endStack[endStack.length - 1]] || stacks[endStack].length === 0) {
> +    movePiece(startStack, endStack);
>
> This does work but see if you can find a way to get it to work without
> having movePiece() inside of isLegal
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/AustinCodingAcademy/javascript-workbook/pull/150#pullrequestreview-21921302>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AU_bcnOlf0n3Fb_QfRW8bWc3_5rF8zTZks5rcnw3gaJpZM4LlJ3v>
> .
>



-- 


*Arnold Wells Photography <http://www.arnoldwellsphotography.com/>Always
Get The Shot!512-844-0404*

Arnold73 committed:
homework done

Arnold73 committed:
mastermind app done

Arnold73 committed:
uploading some completed work

Arnold73 committed:
various stuff

Arnold73 committed:
tic tac toe

Arnold73 committed:
Homework 7 all passing

Arnold73 committed:
tic tac toe final

Arnold73 committed:
final tic tac toe and homework 8

Arnold73 committed:
JQ Towers All tests passing

Arnold73 committed:
06homework and Space

pieroapretto commented on apps/07jquery-tic-tac-toe/script.js:
```diff
@@ -1,10 +1,77 @@
   })
+
+  $('#clear').click(function(){// this clears the board.
+		     $('[data-cell]').text('');
```
Nice shorthand jQuery.

pieroapretto commented on apps/07jquery-tic-tac-toe/script.js:
```diff
@@ -1,10 +1,77 @@
-    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+
+
+    if (checkForWin()){//This checks for the win and announces the winner if the criteria are met.
```
Good shorthand `if` statement.

pieroapretto commented on apps/08jquery-towers-of-hanoi/script.js:
```diff
@@ -1,5 +1,41 @@
+      }
+    }
+    else{
+      block = $(this).children().last().detach();//otherwise block still holds the last block clicked.
```
If you declare `var lastBlockOfTheStack` (line 12) outside your first if statement, you could reuse the logic here. 

`ex: block = lastBlockOfTheStack.detach();`

pieroapretto commented on apps/08jquery-towers-of-hanoi/script.js:
```diff
@@ -1,5 +1,41 @@
+     var StackThreeLength = $('[data-stack="3"]').children().length;
+
+     if(($stackTwoLength === 4) || (StackThreeLength === 4)) {//if any of the non starting stacks = 4 then you won.
+       $('#announce-game-won').text("You Won!");
```
All tests passed. nicely done!

pieroapretto commented on homework/07LessonSeven/script.js:
```diff
@@ -46,7 +46,7 @@ $(document).on('ready', function() {
   // select the elements with class="item"
   // assign the selected jQuery object to the the property window.items
-  window.items = $('item');
+  window.items = $('.item');
```
Good fix.

pieroapretto commented on homework/08LessonEight/script.js:
```diff
@@ -17,30 +17,30 @@ $(document).on('ready', function() {
   // use $finalList.append() to put window.secondListChildren
   // at the end of #final-list
-  $finalList;
+  $finalList.append(window.secondListChildren);
```
All tests passed.

Arnold73 committed:
address bookall tests passed

Arnold73 committed:
Final Project
